### PR TITLE
refactor: Allow admins and instance owners to use any credentials in their workflows

### PR DESCRIPTION
**Pull request by krynble**
## Summary
With the introduction of projects within RBAC we broke part of the functionality which is about allowing instance owners and admins to use any credentials within their workflows.

This PR fixes this.

## Related tickets and issues
https://linear.app/n8n/issue/PAY-1581

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.

### CODE REVIEW

File: packages/cli/src/credentials/credentials.service.ts [hunk 1]
```diff
@@ -95,6 +106,16 @@ export class CredentialsService {
 			return credentials;
 		}
 
+		if (typeof options.listQueryOptions?.filter?.projectId === 'string') {
```
Comment (despairblue) [inline packages/cli/src/credentials/credentials.service.ts:109]:
```suggestion
		// If the workflow is part of a personal project we want to show the credentials the user making the request has access to, not the credentials the user owning the workflow has access to.
		if (typeof options.listQueryOptions?.filter?.projectId === 'string') {
```

**Review by despairblue — COMMENTED** [review]
One of the new tests is failing and when removing the new code no other test fails which could mean that the new logic is not covered. 

I'll check it out and make amends.

**Comment by despairblue** [comment]
Oh and one thing I forgot: I think the contract for this route becomes weird. The FE passes a project ID that the BE swaps out in certain situations, which violates the assumption a consumer may have: I pass `projectId` X and get credentials from project X.

Instead the FE could just pass the workflow ID and then at least the assumption that you're getting back credentials within the project which ID your passing is not violated. Instead the assumption would become: Give me all credentials I'm allowed to use in workflow Z.

File: packages/cli/src/credentials/credentials.service.ts [hunk 2]
```diff
@@ -70,6 +70,17 @@ export class CredentialsService {
...
+				// Only instance owners and admins have the credential:list scope
+				// Those users should be able to use _all_ credentials within their workflows.
+				const projectRelation = projectRelations.find(
+					(relation) => relation.projectId === options.listQueryOptions?.filter?.projectId,
+				);
+				if (projectRelation?.role === 'project:personalOwner') {
+					// Will not affect team projects as these have admins, not owners.
+					delete options.listQueryOptions?.filter?.projectId;
+				}
+			}
```
Comment (despairblue) [inline packages/cli/src/credentials/credentials.service.ts:83]: I believe this is not covered by a test. Is it?

Comment (krynble) [reply packages/cli/src/credentials/credentials.service.ts:83]: You're right, I think it's not, let me quickly add this one! Thanks

**Review by despairblue — APPROVED** [review]

**Pull request merged**